### PR TITLE
Add Updated Gambit

### PR DESCRIPTION
**Pull request by askvortsov1**
**Fixes Part Of flarum/issue-archive#418**

**Changes proposed in this pull request:**
- Add Updated gambit

**Confirmed**

- [x] Frontend changes: tested on a local Flarum installation.
- [x] Backend changes: tests are green (run `composer test`).

### CODE REVIEW

File: src/Discussion/Search/Gambit/UpdatedGambit.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        if (! $search instanceof DiscussionSearch) {
+            throw new LogicException('This gambit can only be applied on a DiscussionSearch');
+        }
+
+        // If we've just been provided with a single YYYY-MM-DD date, then find
+        // discussions that were started on that exact date. But if we've been
+        // provided with a YYYY-MM-DD..YYYY-MM-DD range, then find discussions
+        // that were last updated during that period.
+        if (empty($matches[3])) {
+            $search->getQuery()->whereDate('last_posted_at', $negate ? '!=' : '=', $matches[1]);
```
Comment (franzliedke) [inline src/Discussion/Search/Gambit/UpdatedGambit.php]: Can we ignore (or completely prevent) the negation flag here? I don't really see any use-case, but more importantly, if I am not mistaken, negating would mean that the index on `last_posted_at` could not be used.

**Review by franzliedke — COMMENTED** [review]
Because you're so productive... :grin: 

Can I ask you for a test case for this one? We already have a few integration tests for the search endpoint, if I am not mistaken.

File: src/Discussion/Search/Gambit/UpdatedGambit.php (hunk 1 above)
Comment (askvortsov1) [reply src/Discussion/Search/Gambit/UpdatedGambit.php]: This is pretty much copied from the CreatedGambit. I can def do that, but should I ignore it there as well?

**Comment by askvortsov1** [comment]
> Because you're so productive... 
> 
> Can I ask you for a test case for this one? We already have a few integration tests for the search endpoint, if I am not mistaken.

Absolutely! But before I do that, can we decide whether or not we want this gambit as per https://github.com/flarum/core/issues/256#issuecomment-591916018?

**Comment by datitisev** [comment]
Not stale, just not a priority.

**Comment by askvortsov1** [comment]
Let's keep this around for now

**Pull request closed without merging**